### PR TITLE
matrix voice support

### PR DESCRIPTION
**Pull request by LearnedVector**
* Description of what the PR does, such as fixes # {issue number}
This adds support for Matrix Voice or Matrix Creator.
https://www.matrix.one/

* Description of how to validate or test this PR
follow the auto_run setup prompt to install Matrix Voice.

* Whether you have signed a CLA (Contributor Licensing Agreement)
Yes

### CODE REVIEW

File: home/pi/auto_run.sh [hunk 1]
```diff
@@ -280,7 +280,8 @@ function setup_wizard() {
         echo "  1) PlayStation Eye (USB)"
         echo "  2) Blue Snoball ICE (USB)"
         echo "  3) Google AIY Voice HAT and microphone board (Voice Kit v1)"
-        echo "  4) Other (unsupported -- good luck!)"
+        echo "  4) Matrix Voice HAT."
+        echo "  5) Other (unsupported -- good luck!)"
         echo -n "Choice [1-4]: "
```
Comment (penrods) [inline home/pi/auto_run.sh]: Change to "1-5"

File: home/pi/auto_run.sh [hunk 2]
```diff
@@ -548,6 +566,85 @@ then
...
 
+# Matrix Voice Hat Setup
+if [ -f ~/setup_matrix ]
+then
+    if [ ! -f matrix_setup_state.txt ]
+    then
+        echo ""
+        echo "========================================================================="
+        echo "Setting up Matrix Voice Hat. This will install the matrixio-kernel-modules and pulseaudio"
+        echo "This will require 3 reboots"
```
Comment (penrods) [inline home/pi/auto_run.sh]: echo "This process is automatic, but requires rebooting three times.  Please be patient!"

File: home/pi/auto_run.sh [hunk 3]
```diff
@@ -548,6 +566,85 @@ then
...
+        echo "Press any key to continue..."
+        read -N1 -s anykey
+    else
+        echo "Press any key to continue setting up Matrix Voice HAT"
+        read -N1 -s anykey
+    fi
+
+    if [ ! -f matrix_setup_state.txt ]
+    then
+        echo "Add Matrix repo and key and Update and install packages"
```
Comment (penrods) [inline home/pi/auto_run.sh]: echo "Adding Matrix repo and installing packages..."

No need for all that detail.

File: home/pi/auto_run.sh [hunk 4]
```diff
@@ -548,6 +566,85 @@ then
...
+        # add repo
+        curl https://apt.matrix.one/doc/apt-key.gpg | sudo apt-key add -
+        echo "deb https://apt.matrix.one/raspbian $(lsb_release -sc) main" | sudo tee /etc/apt/sources.list.d/matrixlabs.list
+        sudo apt-get update -y
+        sudo apt-get upgrade -y
+
+        echo "stage-1" > matrix_setup_state.txt
+        echo "Needs to reboot in case of kernel updates"
```
Comment (penrods) [inline home/pi/auto_run.sh]: echo "Rebooting to apply kernel updates, the installation will resume afterwards"

File: home/pi/auto_run.sh [hunk 5]
```diff
@@ -548,6 +566,85 @@ then
...
+        read -p "Press enter to continue reboot"
+        sudo reboot
+    else
+        matrix_setup_state=$( cat matrix_setup_state.txt)
+    fi
+
+    if [ $matrix_setup_state == "stage-1" ]
+    then
+        echo "installing matcixio-kernel-modules"
```
Comment (penrods) [inline home/pi/auto_run.sh]: echo "Installing matrixio-kernel-modules..."

File: home/pi/auto_run.sh [hunk 6]
```diff
@@ -548,6 +566,85 @@ then
...
+        sudo apt install matrixio-kernel-modules -y
+
+        echo "installing pulseaudio"
+        sudo apt-get install pulseaudio -y
+        
+        echo "Need to reboot after installing matcixio-kernel-modules and pulseaudio"
```
Comment (penrods) [inline home/pi/auto_run.sh]: echo "Rebooting to apply audio subsystem changes, the installation will continue afterwards."

File: home/pi/auto_run.sh [hunk 7]
```diff
@@ -548,6 +566,85 @@ then
...
+        read -p "Press enter to continue reboot"
+        echo "stage-2" > matrix_setup_state.txt
+        sudo reboot
+    fi
+
+    if [ $matrix_setup_state == "stage-2" ]
+    then
+        echo "setting Matrix as standard microphone..."
```
Comment (penrods) [inline home/pi/auto_run.sh]: echo "Setting ...

File: home/pi/auto_run.sh [hunk 8]
```diff
@@ -548,6 +566,85 @@ then
...
+        amixer
+        sleep 2
+        amixer
+
+        mycroft-mic-test
+        
+        read -p "You should have heard the recording playback. Press enter to continue"
+
+        echo "========================================================================="
+        echo "updating the python virtual environment"
```
Comment (penrods) [inline home/pi/auto_run.sh]: echo "Updating

File: home/pi/auto_run.sh [hunk 9]
```diff
@@ -548,6 +566,85 @@ then
...
+        bash mycroft-core/dev_setup.sh
+
+        echo "stage-3" > matrix_setup_state.txt
+        read -p "Press enter to continue reboot! this should be the last one"
```
Comment (penrods) [inline home/pi/auto_run.sh]: read -p "Your Matrix microphone is now setup!  Press enter to perform the final reboot and start Mycroft."

**Review by penrods — CHANGES_REQUESTED** [review]
Looking good!  I tossed in some edits of prompts, nothing major.  I'd like to walk through the process myself, also, to see how it feels.

**Comment by LearnedVector** [comment]
@penrods updated the prompts ^^

**Comment by LexiconCode** [comment]
I just wanted to show my support. I'm looking forward to merge :) @penrods

**Pull request merged**